src/client/render/equirectangular.cpp

```diff
@@ -0,0 +1,212 @@
+
+void RenderingCoreEquirectangular::initTextures()
+{
+	RenderingCoreCubeMap::initTextures(cubemap_size);
+
+	renderOutput = driver->addRenderTargetTexture(
+			output_size, "3d_render_eq", video::ECF_A8R8G8B8);
+}
+
+void RenderingCoreEquirectangular::clearTextures()
+{
+	RenderingCoreCubeMap::clearTextures();
+
+	driver->removeTexture(renderOutput);
+}
+
+void RenderingCoreEquirectangular::processImages()
+{
+	u32 *pixel = (u32 *) renderOutput->lock();
+	u32 *pixelf[6];
+
+	for (int i = 0; i < 6; i ++)
+		pixelf[i] = (u32 *) faces[i]->lock();
+
+	u32 dims = output_size.Width / 4;
+	for (u32 j = 0; j < dims * 2; j ++) {
+		double v = 1.0 - (double) j / (dims * 2);
+		double phi = v * core::PI;
+
+		for (u32 i = 0; i < dims * 4; i ++) {
+			double u = (double) i / (dims * 4);
+			double theta = u * 2 * core::PI;
+
+			double x = sin(theta) * sin(phi);
+			double y = cos(phi);
+			double z = cos(theta) * sin(phi);
+
+			double a = fmax(fmax(abs(x), abs(y)), abs(z));
+
+			double xx = x / a;
+			double yy = y / a;
+			double zz = z / a;
+
+			u32 xPixel = 0;
+			u32 yPixel = 0;
+			u32 yTemp = 0;
+			int selected_face = -1;
+
+			if (xx == 1) {
+				xPixel = ((-1 * tan(atan(z / x)) + 1.0) / 2.0) * dims;
+				yTemp = ((tan(atan(y / x)) + 1.0) / 2.0) * dims;
+				selected_face = 1; // X-
+			} else if (xx == -1) {
+				xPixel = ((-1 * tan(atan(z / x)) + 1.0) / 2.0) * dims;
+				yTemp = ((-1 * tan(atan(y / x)) + 1.0) / 2.0) * dims;
+				selected_face = 0; // X+
+			} else if (yy == 1) {
+				xPixel = ((tan(atan(x / y)) + 1.0) / 2.0) * dims;
+				yTemp = ((-1 * tan(atan(z / y)) + 1.0) / 2.0) * dims;
+				selected_face = 2; // Y+
+			} else if (yy == -1) {
+				xPixel = ((-1 * tan(atan(x / y)) + 1.0) / 2.0) * dims;
+				yTemp = ((-1 * tan(atan(z / y)) + 1.0) / 2.0) * dims;
+				selected_face = 3; // Y-
+			} else if (zz == 1) {
+				xPixel = ((tan(atan(x / z)) + 1.0) / 2.0) * dims;
+				yTemp = ((tan(atan(y / z)) + 1.0) / 2.0) * dims;
+				selected_face = 5; // Z-
+			} else if (zz == -1) {
+				xPixel = ((tan(atan(x / z)) + 1.0) / 2.0) * dims;
+				yTemp = ((-1 * tan(atan(y / z)) + 1.0) / 2.0) * dims;
+				selected_face = 4; // Z+
+			}
+
+			xPixel = fmin(xPixel, dims - 1);
+			yPixel = fmin(yTemp, dims - 1);
+			yPixel = dims - 1 - yPixel;
+
+			if (selected_face == 2 || selected_face == 3) {
+				pixel[j * dims * 4 + i] = pixelf[selected_face][xPixel * dims + (dims - 1 - yPixel)];
+			} else {
+				pixel[j * dims * 4 + i] = pixelf[selected_face][yPixel * dims + xPixel];
+			}
+		}
+	}
+
+	for (int i = 0; i < 6; i ++)
+		faces[i]->unlock();
+
+	renderOutput->unlock();
+}
+
+void RenderingCoreEquirectangular::drawAll()
+{
+	driver->setRenderTarget(nullptr, false, false, skycolor);
+	draw3D();
+	drawHUD();
+
+	if (!client->is360VideoMode())
+		return;
+
+	camera->m_enable_draw_wielded_tool = false;
+	for (int i = 0; i < 6; i ++) {
+		useFace(i);
+		draw3D();
+	}
+	camera->m_enable_draw_wielded_tool = true;
+
+	processImages();
+
+	if (saveAsImage) {
```

Comment:
if will be very write intensive to do no ?

Reply:
Yes, it is. But, if you don't want to use it, you can just use screen recording app. Any other idea to do it internally?

The main reason I add this feature is because the issue's author said it,
> I would imagine it could be challenging to capture equirectangular video with standard capture software, so if Minetest could render the video to disk (mJPEG would be big files but very light on the CPU, so would be my recommendation) that'd be even better.